Update target platforms

briandealwis opened the pull request:
- Photon was released today
- Update Oxygen to SR3
- WTP seems to have removed their old releases from Neon and prior, so the SDK features aren't available

briandealwis commented:
WTP SDKs are available from the [archives page](http://archive.eclipse.org/webtools/downloads/drops/?d) but since we're defaulting to Oxygen, I'm not going to bother.

elharo commented:
I had hoped we'd get one last Neon release before dropping it, but if it's too much trouble c'est la vie.

chanseokoh commented:
Neon builds still failing... the SDK seems required.
```
36812 [ERROR] Cannot resolve project dependencies:
36812 [ERROR]   Software being installed: com.google.cloud.tools.eclipse.dist 1.7.0.qualifier
36812 [ERROR]   Missing requirement: com.google.cloud.tools.eclipse.appengine.localserver 0.1.0.qualifier requires 'bundle org.eclipse.jst.server.ui 1.1.300' but it could not be found
```

elharo commented:
Not quite sure what's up here; but it seems our build depends on remote resources that do not have stable locations? If so, should we check those into our repo instead?

chanseokoh commented:
As I understand, the SDK for Neon is gone from the usual location but still available after moved to the archive location (https://github.com/GoogleCloudPlatform/google-cloud-eclipse/pull/3184#issuecomment-400826422).

We thought we wouldn't bother about pulling the SDK in for Neon, thinking it is only for retrieving source code and we don't target Neon in our dev environment. However, seems like we're wrong and need to pull in the SDK from the archive.

briandealwis commented:
I incorrectly thought we were pulling in the WTP dependencies from the release train locations (download.eclipse.org/releases/neon).  But we pull in all of the WTP dependencies using the SDK features.  I've updated the Neon TP to use the archive site for Neon.  I also updated SWTBot to the latest.
